src/libs/actions/Policy/ReportField.ts

```diff
@@ -168,24 +169,46 @@ function createReportField(policyID: string, {name, type, initialValue}: CreateR
         fieldID,
         orderWeight: Object.keys(previousFieldList).length + 1,
         deletable: false,
-        value: type === CONST.REPORT_FIELD_TYPES.LIST ? CONST.REPORT_FIELD_TYPES.LIST : null,
         keys: [],
         externalIDs: [],
         isTax: false,
     };
-    const onyxData: OnyxData = {
-        optimisticData: [
-            {
-                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
-                onyxMethod: Onyx.METHOD.MERGE,
-                value: {
-                    fieldList: {
-                        [fieldKey]: {...newReportField, pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
-                    },
-                    errorFields: null,
+
+    const optimisticReportFieldDataForPolicy: OnyxValueWithOfflineFeedback<PolicyReportField> = {
+        ...newReportField,
+        value: type === CONST.REPORT_FIELD_TYPES.LIST ? CONST.REPORT_FIELD_TYPES.LIST : null,
+    };
+
+    const policyExpenseReports = Object.values(ReportConnection.getAllReports() ?? {}).filter(
+        (report) => report?.policyID === policyID && report.type === CONST.REPORT.TYPE.EXPENSE,
+    ) as Report[];
+
+    const onyxOperationsToAddExpenseReportFields = policyExpenseReports.map((report) => ({
+        key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+        onyxMethod: Onyx.METHOD.MERGE,
+        value: {
+            fieldList: {
+                [fieldKey]: {...newReportField, pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
+            },
+        },
+    }));
+
+    const optimisticData = [
+        ...onyxOperationsToAddExpenseReportFields,
+        {
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
```

Comment:
We need to remove it in the failure data too now. The pending RBR is going to be on the report fields in settings